### DISCUSSION TITLE
✅ Connect to ssh as SOCKS client (ssh -D)

### DISCUSSION THREAD

**Discussion by ghost** [isontheline/pro.webssh.net#606] [Ideas]
A great addition would be allowing opening the ssh tunnel as a proxy server for a SOCKS client (equivalent of ssh -D).

It would allow users to connect to and route all traffic through their home server when abroad to access local services blocked outside the country. Basically a free VPN!

I can't see this is possible on the current version, but please let me know if I've missed something.

Many thanks,

Paul.

**Comment by isontheline** [comment]
Hello @blendexperts Paul 👋

Do you need to be able to have your iPhone / iPad / Mac to accept SOCKS connections from your LAN?
or
Do you want to use other apps on your iPhone / iPad to access the SOCKS server?

The second option is available by using DPF : `*:9999` where 9999 is a chopped listening port

The first option isn't yet available but it seems I have planned it :)
![image](https://user-images.githubusercontent.com/44212923/165595767-e783bf8f-c7ba-41da-a114-c3db8f0c3b47.png)

Which option are you talking about?

**Comment by ghost** [comment]
It is the former. I'd like to be able to set a proxy configuration on the client network connection to route all traffic via SOCKS to my SSH server at home.

**Reply by isontheline** [reply]
Thanks for your feedback!

I have opened an issue in order to track progress : https://github.com/isontheline/pro.webssh.net/issues/607

**Reply by isontheline** [reply]
Hi @blendexperts 👋

I have updated WebSSH in order to enable SOCKS server : #607

**Comment by ash34** [comment]
How do I use this? I enter the SSH server as the proxy address in my SOCKS client? Which port should be quoted?

**Reply by isontheline** [reply]
Enter `*:1985` inside the port forward rules

`1985` is the port you will use inside your SOCKS client. The IP address should be 127.0.0.1 if using on your device or use your LAN IP.

Don't forget to allow incoming connections :
Allow WebSSH to accept external connections : iOS Settings > WebSSH > Tunnel > Incoming Connections Strategy > Accept both local and remote connections
